xbmc/video/VideoDatabase.cpp

```diff
@@ -4964,11 +5104,96 @@ void CVideoDatabase::UpdateTables(int iVersion)
     m_pDS->exec("ALTER TABLE musicvideo ADD premiered TEXT");
     m_pDS->exec(PrepareSQL("UPDATE musicvideo SET premiered=c%02d", VIDEODB_ID_MUSICVIDEO_YEAR));
   }
+
+  if (iVersion < 107)
+  {
+    // need this due to the nested GetScraperPath query
+    std::unique_ptr<Dataset> pDS;
+    pDS.reset(m_pDB->CreateDataset());
+    if (NULL == pDS.get()) return;
+
+    pDS->exec("CREATE TABLE uniqueid (uniqueid_id INTEGER PRIMARY KEY, media_id INTEGER, media_type TEXT, value TEXT, type TEXT)");
+
+    std::map<int, std::string> pathTypes;
+    for (int i = 0; i < 3; ++i)
+    {
+      std::string query, table, column;
+      switch (i)
+      {
+      case (0):
+        query = PrepareSQL("SELECT DISTINCT movie.idMovie, movie.c%02d, path.idPath, path.strPath FROM movie "
+                           "  JOIN files ON files.idFile=movie.idFile "
+                           "  JOIN path ON path.idPath=files.idPath", VIDEODB_ID_IDENT_ID);
+        table = "movie";
+        column = "idMovie";
+        break;
+      case (1):
+        query = PrepareSQL("SELECT tvshow.idShow, tvshow.c%02d, path.idPath, path.strPath FROM tvshow "
+                           "  JOIN tvshowlinkpath ON tvshowlinkpath.idShow=tvshow.idShow "
+                           "  JOIN path ON path.idPath=tvshowlinkpath.idPath "
+                           "GROUP BY tvshow.idShow", VIDEODB_ID_TV_IDENT_ID);
+        table = "tvshow";
+        column = "idShow";
+        break;
+      case (2):
+        query = PrepareSQL("SELECT DISTINCT episode.idEpisode, episode.c%02d, path.idPath, path.strPath FROM episode "
+                           "  JOIN files ON files.idFile=episode.idFile "
+                           "  JOIN path ON path.idPath=files.idPath", VIDEODB_ID_EPISODE_IDENT_ID);
+        table = "episode";
+        column = "idEpisode";
+        break;
+      default:
+        continue;
+      }
+      pDS->query(query);
+      while (!pDS->eof())
+      {
+
+        std::string uniqueid = pDS->fv(1).get_asString();
+        if (!uniqueid.empty())
+        {
+          std::string type = "unknown";
+          const auto& pathType = pathTypes.find(pDS->fv(2).get_asInt());
+          if (pathType == pathTypes.end())
+          {
+            if (StringUtils::StartsWith(uniqueid, "tt"))
+            {
+              type = "imdb";
+            }
+            else
+            {
+              ADDON::ScraperPtr scraper = GetScraperForPath(pDS->fv(3).get_asString());
+              if (scraper != nullptr)
+              {
+                if (scraper->ID() == "metadata.themoviedb.org" || scraper->ID() == "metadata.universal")
```

Comment:
Why not do the simple thing use the above 'unknown' type for everything? This whole upgrade guesswork looks very fragile. Just because some scraper is set for a path does not mean the ids are from that scraper.